src/components/Autocomplete.js

```diff
@@ -67,157 +37,174 @@ export default class Autocomplete extends React.Component {
-			this.lookup = new SmartySDK.usAutocompletePro.Lookup(query);
-			this.client = this.autoCompleteClient;
-			if (hasSecondaries) {
-				this.lookup.selected = query;
-			}
-		} else {
-			this.client = this.internationalautocompleteClient;
-			if (hasSecondaries) {
-				this.lookup = new SmartySDK.internationalAddressAutocomplete.Lookup({addressId: addressId});
-			} else {
-				this.lookup = new SmartySDK.internationalAddressAutocomplete.Lookup(query);
-				this.lookup.search = query;
-			}
+	// Helper
+	const createUSLookup = (query, hasSecondaries) => {
+		const lookup = new SmartySDK.usAutocompletePro.Lookup(query);
+		if (hasSecondaries) {
+			lookup.selected = query;
 		}
-		this.lookup.country = this.state.country;
-
-		this.client.send(this.lookup)
+		return lookup;
+	};
+	
+	// Helper
+	const createInternationalLookup = (query, addressId, hasSecondaries) => {
+		const lookup = new SmartySDK.internationalAddressAutocomplete.Lookup(hasSecondaries ? { addressId } : query);
+		if (!hasSecondaries) {
+			lookup.search = query;
+		}
+		return lookup;
+	};
+	
+	const queryAutocompleteForSuggestions = (query, addressId, hasSecondaries = false) => {
+		const isUS = formValues.country === "US";
+		
+		const client = isUS ? autoCompleteClient : internationalautocompleteClient;
+		const lookup = isUS
+			? createUSLookup(query, hasSecondaries)
+			: createInternationalLookup(query, addressId, hasSecondaries);
+	
+		lookup.country = formValues.country;
+	
+		client.send(lookup)
 			.then(results => {
-				this.setState({suggestions: results});
+				setSuggestions(results);
 			})
 			.catch(console.warn);
-	}
+	};
 
-	selectSuggestion(suggestion) {
-		if (suggestion.entries > 1) {
-			this.queryAutocompleteForSuggestions(this.formatAutocompleteSuggestion(suggestion), suggestion.addressId, true);
-		} else {
-			this.useAutoCompleteSuggestion(suggestion)
-				.then(() => {
-					if (this.state.shouldValidate) this.validateAddress();
-				});
+	const getFormValues = (address) => {
+		if (address.addressText) {
+			return {
+				...formValues,
+				freeform: address.addressText,
+			}
+		}
+
+		return {
+			...formValues,
+			address1: address.streetLine,
+			address2: address.secondary,
+			city: address.city,
+			state: address.state,
+			zipCode: address.zipcode,
 		}
 	}
 
-	useAutoCompleteSuggestion(suggestion) {
-		if (this.state.country === "US") {
-			return new Promise(resolve => {
-				this.setState({
-					address1: suggestion.streetLine,
-					address2: suggestion.secondary,
-					city: suggestion.city,
-					state: suggestion.state,
-					zipCode: suggestion.zipcode,
-					suggestions: {result: []},
-				}, resolve);
-			});
+	const useAutoCompleteSuggestion = (suggestion) => {
+		if (formValues.country === "US") {
+			setFormValues(getFormValues(suggestion));
 		} else {
-			return new Promise(resolve => {
-				this.setState({
-					freeform: suggestion.addressText,
-				}, resolve);
-			})
+			setFreeform(suggestion.addressText);
```

Comment:
I noticed that the `freeform` state is unused and so I'm wondering if this needs to be here either? It seems like `freeform` is manually added to the address object in `getFormValues` when it's needed so maybe it's not needed here either?

Reply:
Oh, yep, you're right! I missed this. Good catch